lib/job/arSearchResultExportCsvJob.class.php

```diff
@@ -0,0 +1,235 @@
+
+    if ($success !== true)
+    {
+      $this->error('Failed to create ZIP file.');
+      return false;
+    }
+
+    // Mark job as complete and set download path
+    $this->info('Export and archiving complete.');
+    $this->job->setStatusCompleted();
+    $this->job->downloadPath = $this->getDownloadRelativeFilePath();
+    $this->job->save();
+
+    return true;
+  }
+
+  /**
+   * Add criteria to query based on parameters.
+   *
+   * @return void
+   */
+  protected function addCriteriaBasedOnSearchParameters()
+  {
+    // Add criteria for main search fields
+    if (null !== $criterias = $this->parseQuery())
+    {
+      $this->search->queryBool->addMust($criterias);
+    }
+
+    // Add criteria fo secondary search fields
+    foreach (SearchAdvancedAction::$NAMES as $name)
+    {
+      if (
+        !empty($this->searchParams[$name])
+        && (null !== $criterias = SearchAdvancedAction::fieldCriteria($name, $this->searchParams[$name]))
+      )
+      {
+        $this->search->queryBool->addMust($criterias);
+      }
+    }
+
+    // Set query if criteria were added
+    if (count($this->search->queryBool->getParams()))
+    {
+      $this->search->query->setQuery($this->search->queryBool);
+    }
+  }
+
+  /**
+   * Translate array of search parameters to query criteria.
+   *
+   * Modified version of parseQuery method in the SearchAdvancedAction class
+   *
+   * Each set of parameters is numbered, starting at zero, and includes three
+   * properties: query text (prefixed by "sq"), operation (prefixed by "so": "and" or
+   * "or"), and fields (prefixed by "sf") to return (defaulting to "_all").
+   *
+   * For example:
+   *
+   *   $this->searchParams = array(
+   *     'so0' => 'and',
+   *     'sq0' => 'cats',
+   *     'sf0' => ''
+   *   );
+   *
+   * @return object  \Elastica\Query\Bool instance
+   */
+  protected function parseQuery()
+  {
+    $queryBool = new \Elastica\Query\Bool();
+
+    $count = -1;
+
+    while (null !== $query = $this->searchParams['sq'.++$count])
+    {
+      if (empty($query)) continue;
+
+      $field = $this->searchParams['sf'.$count];
+      if (empty($field))
+      {
+        $field = '_all';
+      }
+
+      $operator = $this->searchParams['so'.$count];
+      if (empty($operator))
+      {
+        $operator = 'or';
+      }
+
+      $queryField = SearchAdvancedAction::queryField($field, $query, $this->archivalStandard);
+      SearchAdvancedAction::addToQueryBool($queryBool, $operator, $queryField);
+    }
+
+    if (0 == count($queryBool->getParams()))
+    {
+      return;
+    }
+
+    return $queryBool;
+  }
+
+  /**
+   * Export search results as CSV
+   *
+   * @param string  Path of file to write CSV data to
+   *
+   * @return int  Number of descriptions exported
+   */
+  protected function exportResults($path)
+  {
+    $itemsExported = 0;
+
+    // Exporter will create a new file each 10,000 rows
+    $writer = new csvInformationObjectExport(
```

Comment:
I think it's fine to put this all in one line as long as the line is around < 120 chars